src/Microsoft.CodeQuality.Analyzers/Core/ApiDesignGuidelines/StringBuilderAppendShouldNotTakeSubstring.cs

```diff
@@ -0,0 +1,150 @@
+
+        private static readonly LocalizableString s_localizableTitle2 = new LocalizableResourceString(
+            nameof(MicrosoftApiDesignGuidelinesAnalyzersResources.StringBuilderShouldUseSubstringOverloadWithTwoParameterTitle),
+            MicrosoftApiDesignGuidelinesAnalyzersResources.ResourceManager, 
+            typeof(MicrosoftApiDesignGuidelinesAnalyzersResources));
+        private static readonly LocalizableString s_localizableMessageDefault2 = new LocalizableResourceString(
+            nameof(MicrosoftApiDesignGuidelinesAnalyzersResources.StringBuilderShouldUseSubstringOverloadWithTwoParameterDefault),
+            MicrosoftApiDesignGuidelinesAnalyzersResources.ResourceManager,
+            typeof(MicrosoftApiDesignGuidelinesAnalyzersResources));
+        private static readonly LocalizableString s_localizableDescription2 = new LocalizableResourceString(
+            nameof(MicrosoftApiDesignGuidelinesAnalyzersResources.StringBuilderShouldUseSubstringOverloadWithTwoParameterDescription),
+            MicrosoftApiDesignGuidelinesAnalyzersResources.ResourceManager, 
+            typeof(MicrosoftApiDesignGuidelinesAnalyzersResources));
+
+        internal static DiagnosticDescriptor RuleReplaceOneParameter = new DiagnosticDescriptor(
+            RuleIdOneParameterId,
+            s_localizableTitle,
+            s_localizableMessageDefault,
+            DiagnosticCategory.Usage,
+            DiagnosticHelpers.DefaultDiagnosticSeverity,
+            isEnabledByDefault: DiagnosticHelpers.EnabledByDefaultIfNotBuildingVSIX,
+            description: s_localizableDescription,
+            helpLinkUri: "", // TODO (might require an addition to docs repository?
+            customTags: WellKnownDiagnosticTags.AnalyzerException);
+
+        internal static DiagnosticDescriptor RuleReplaceTwoParameter = new DiagnosticDescriptor(
+            RuleIdTwoParameterId,
+            s_localizableTitle2,
+            s_localizableMessageDefault2,
+            DiagnosticCategory.Usage,
+            DiagnosticHelpers.DefaultDiagnosticSeverity,
+            isEnabledByDefault: DiagnosticHelpers.EnabledByDefaultIfNotBuildingVSIX,
+            description: s_localizableDescription2,
+            helpLinkUri: "", // TODO (might require an addition to docs repository?
+            customTags: WellKnownDiagnosticTags.AnalyzerException);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(RuleReplaceOneParameter, RuleReplaceTwoParameter);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+
+            context.RegisterCompilationStartAction(this.OnCompilationStart);
+        }
+
+        private void OnCompilationStart(CompilationStartAnalysisContext startContext)
+        {
+            var compilation = startContext.Compilation;
```

Comment:
>var compilation [](start = 12, length = 15)

Nit: Don't really need this local.

Reply:
it's used twice, Compilation is a property, so accessing the property twice compares to an additional local variable - not sure what's better, so I changed it as requested.